src/gamelog.go

```diff
@@ -19,5 +28,40 @@ func tailLog(filename string) ([]string, error) {
 		result = append(result, line.Text)
 	}
 
-	return result, nil
+  if reformatTimestampsEnabled {
+    result = reformatTimestamps(result)
+  }
+
+  return result, nil
+}
+
+func getStartTime(log []string) {
+  re, _ := regexp.Compile(`\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}`)
+  date := string(re.FindString(log[0]))
+  startTime, _ = time.Parse(time.RFC3339, strings.Replace(date, " ", "T", 1) + "Z")
+}
+
+// convert an offset timestamp from the log to a human readable timestamp
+func offsetToTimestamp(offset string) (string) {
+    offsetDuration, err := time.ParseDuration(offset + "s")
+    timestamp := startTime.Add(offsetDuration)
+
+    if err == nil  {
+      return timestamp.Format("2006-01-02 15:04:05") +  ":" + strings.Replace(offset, offset, "", 1)
+    } else {
+      return offset
+    }
+}
+
+func reformatTimestamps(log []string) ([]string) {
+  getStartTime(log)
+  result := []string{}
+
+  for i := range log {
```

Comment:
use key-value for loop: `for index, line := range log {` or in this case `for _, line := range log {`, since you dont need the index.